### PR TITLE
Ci/publish staging

### PR DESCRIPTION
**Pull request by hleonardtek**
The purpose is to deploy on staging whenever the main branch is pushed.
I put the deploy job in the publish workflow to avoid duplication.

### CODE REVIEW

**Review by fredericvilcot — APPROVED** [review]

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -22,7 +22,7 @@ jobs:
         with:
           images: ghcr.io/${{ github.repository }}
           tags: |
-            type=raw,enable=${{ endsWith(github.ref, github.event.repository.default_branch) }},value=nightly
+            type=raw,enable=${{ endsWith(github.ref, github.event.repository.default_branch) }},value=nightly-${{ github.sha }}
```
Comment (amimart) [inline .github/workflows/publish.yml:25]: No please, do not touch the tags, we'll use the nightly.

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -44,3 +44,27 @@ jobs:
...
+
+      - name: Set up kubernetes config
+        run: |
+          echo ${{ secrets.KUBECONFIG }} > config_enc
+          cat config_enc | base64 -d > config
+          rm config_enc
+
+      - name: Deploy staging
+        run: |
+          kubectl set image deployment/dataverse-portal dataverse-portal=${DATAVERSE_PORTAL_IMAGE}:nightly-${{ github.sha }} -n staging
```
Comment (amimart) [inline .github/workflows/publish.yml:70]: We just need to update a metadata in order to trigger the rolling update:
```sh
kubectl patch deploy deploy dataverse-portal -p '{metadata:{"labels": {"commit": "${{ github.sha }}"}}}'
```

File: .github/workflows/publish.yml [hunk 3]
```diff
@@ -44,3 +44,27 @@ jobs:
           push: true
           tags: ${{ steps.docker_metadata.outputs.tags }}
           labels: ${{ steps.docker_metadata.outputs.labels }}
+
+  deploy-k8s-staging:
+    runs-on: ubuntu-22.04
+    needs:
+      - publish-docker-images
```
Comment (amimart) [inline .github/workflows/publish.yml:51]:
```suggestion
      - publish-docker-images
      if: github.ref == 'refs/heads/main'
```

We can condition the deploy to main pushes only.

**Review by amimart — CHANGES_REQUESTED** [review]
Thanks, it works but I'll prefer avoid publishing new tags, we need to keep the nightly tag as is.

Instead, we only need to trigger the rolling update by updating a label on the Deployment.

File: .github/workflows/publish.yml (hunk 2 above)
Comment (hleonardtek) [reply .github/workflows/publish.yml:70]: Ok I just understood what you meant :open_mouth: .
Very smart thanks no need to change the tag  :100: !

**Review by ErikssonJoakim — APPROVED** [review]

File: .github/workflows/publish.yml (hunk 2 above)
Comment (amimart) [reply .github/workflows/publish.yml]: Yes there is 😆

**Review by amimart — APPROVED** [review]
Looks good! 👍

**Review by ccamel — APPROVED** [review]
Great! :+1:

**Comment by hleonardtek** [comment]
@amimart @ccamel I forgot the namespace in the patch command.

**Review by ccamel — APPROVED** [review]

**Review by amimart — APPROVED** [review]

**Review by ccamel — APPROVED** [review]

**Pull request merged**